# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/test/xpcshell/telemetry/test_qm_first_initialization_attempt.js
# Commit: b84880c5de44
# Full Hash: b84880c5de443e6274953d210444c57319ce4b5f
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-13 09:18:36
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Make it possible to initialize temporary origins without ensuring origin directories; r=dom-storage-reviewers,jari
#   
#   LSNG already uses some QuotaManager APIs to achieve that origin directories are
#   not created if they don't exist during datastore preparation, but the feature
#   is not easy to use and it's also not generalized enough for use in other quota
# ==============================================================================

diff -r c5f28ed5382b -r b84880c5de44 dom/quota/test/xpcshell/telemetry/test_qm_first_initialization_attempt.js
--- a/dom/quota/test/xpcshell/telemetry/test_qm_first_initialization_attempt.js	Sat Oct 12 23:42:06 2024 +0300
+++ b/dom/quota/test/xpcshell/telemetry/test_qm_first_initialization_attempt.js	Sat Oct 12 20:58:36 2024 +0000
@@ -594,7 +594,11 @@
       },
       {
         name: initTemporaryOrigin,
-        args: ["default", getPrincipal("https://example2.com")],
+        args: [
+          "default",
+          getPrincipal("https://example2.com"),
+          /* createIfNonExistent */ true,
+        ],
       },
     ],
     expectedSnapshots: {
@@ -668,15 +672,27 @@
     initFunctions: [
       {
         name: initTemporaryOrigin,
-        args: ["temporary", getPrincipal("https://example.com")],
+        args: [
+          "temporary",
+          getPrincipal("https://example.com"),
+          /* createIfNonExistent */ true,
+        ],
       },
       {
         name: initTemporaryOrigin,
-        args: ["default", getPrincipal("https://example.com")],
+        args: [
+          "default",
+          getPrincipal("https://example.com"),
+          /* createIfNonExistent */ true,
+        ],
       },
       {
         name: initTemporaryOrigin,
-        args: ["default", getPrincipal("https://example1.com")],
+        args: [
+          "default",
+          getPrincipal("https://example1.com"),
+          /* createIfNonExistent */ true,
+        ],
       },
       {
         name: initPersistentOrigin,